### ISSUE TITLE
New file and new folder keymaps are broken in January 2019 update

### ISSUE THREAD

**Issue by torrick** [Microsoft/vscode-atom-keybindings#56]
https://code.visualstudio.com/updates/v1_31#_new-tree-widget

Version: 1.31.0

Expected: `a` key opens dialog for new file, `Shift+a` opens up dialog for new folder
Actual: Both result in triggering the tree widget filter.

https://github.com/Microsoft/vscode/issues/68083

**Comment by Ky6uk** [comment]
🤔 I tried to `vscode.commands.executeCommand('setContext', 'listAutomaticKeyboardNavigation', false);` in this, but it has no effect. I have no idea why.

**Comment by YannMCX** [comment]
Hello @torrick @Ky6uk ,
You can fix this by adding this in the VSCode configuration :
```json
"workbench.list.keyboardNavigation": "simple",
```

But a problem occurs with the focus when creating the file or folder, it is created at the root of the project. I couldn't fix that.

And, of course, by changing the setting, you lose access to the new added function 😄

**Comment by Ky6uk** [comment]
@YannMCX Unfortunately setting `"simple"` [doesn't work](https://github.com/Microsoft/vscode/issues/68083).

Also there is no losing access to the new functionality because we're able to add another hotkey to start filtration (toggling context). Look [at PR](https://github.com/Microsoft/vscode-atom-keybindings/pull/57), there is the key <kbd>/</kbd>.

**Comment by auchenberg** [comment]
New release published, https://github.com/Microsoft/vscode-atom-keybindings/releases/tag/v3.0.6

**Comment by jack-sf** [comment]
So if I've this extension in 3.0.6 installed, it should just work?

Because atm it still doesn't work for me: https://www.dropbox.com/s/bjwbpa6ej30aviq/Screenshot%202019-02-11%2011.06.45.png?dl=0 . 

Keyboard navigation in file explorer is triggered automatically with pressing any letter (no `/` prefix needed). Thus, I still can't use shortcuts like `a` for adding new file.

Any tips what should I do?

**Comment by Ky6uk** [comment]
> So if I've this extension in 3.0.6 installed, it should just work?

@jack-sf no, it doesn't. Currently it's broken.

**Comment by telmen** [comment]
Any update on this? @auchenberg

**Comment by joaomoreno** [comment]
@jack-sf @Ky6uk @telmen 

I've just installed 3.0.6 and I've confirmed that the tree filter functionality is only triggered when pressing <kbd>/</kbd>. Do you still see the issue?

**Comment by astorath** [comment]
> @jack-sf @Ky6uk @telmen
> 
> I've just installed 3.0.6 and I've confirmed that the tree filter functionality is only triggered when pressing /. Do you still see the issue?

Yes, the issue is still  present but not at the start (in a few hours of work). Had to rollback to 1.30.2 for a while.

**Comment by Ky6uk** [comment]
@joaomoreno I still see [this](https://github.com/Microsoft/vscode/issues/68161) issue as I did mention it before.

**Comment by jack-sf** [comment]
Yeah. I still got the issue, I have atom-keybindings 3.0.6 and VSCode 1.31.1 . I also have the issue with file losing focus in explorer. 

Let me know guys if I can help you debug/fix it anyhow? (Although I've never done anything in VSCode extensions code yet.)

**Comment by joaomoreno** [comment]
> Yeah. I still got the issue

@jack-sf, is is just like @astorath that _the issue is still present but not at the start (in a few hours of work)_?

> I also have the issue with file losing focus in explorer.

What issue is that?

**Comment by Ky6uk** [comment]
@jack-sf @telmen it seems problem won't be fixed without changes in VSCode because keybindings for `explorer.*` commands are just ignoring (https://github.com/Microsoft/vscode/issues/68960). 😞

**Comment by joaomoreno** [comment]
@Ky6uk You commit should've fixed it: https://github.com/Microsoft/vscode-atom-keybindings/commit/03a70d29e62bd1e1b52ca7277c98e51d4350f4c5 Did you test it when you pushed it? I did, and it fixed on my side. Did it not fix the issue for you?

**Comment by Ky6uk** [comment]
@joaomoreno No, it's not fixed for me. what is "fixed" you mean? Can you create new file or new directory using hotkeys <kbd>A</kbd> and <kbd>Shift+A</kbd>? Because I can't. The latest insider build of VSCode just ignores those hotkeys (`v3.0.6` of this plugin is installed, keyboard navigation: **simple**, no other settings/plugins are installed).

**Comment by joaomoreno** [comment]
You already know your issue will be fixed by https://github.com/Microsoft/vscode/issues/68233 That is completely independent from atom keybindings.

Atom keybindings users can use list navigation keybindings because of your fix in this repository: https://github.com/Microsoft/vscode-atom-keybindings/pull/57.

When I asked _Did it not fix the issue for you?_, I was referring to: can you use the atom keybindings list keyboard navigation actions?

**Comment by Ky6uk** [comment]
@joaomoreno 
> can you use the atom keybindings list keyboard navigation actions?

Sorry, maybe I didn't understand properly, but what does it mean "keyboard navigation actions"? 

Because it looks in this thread all of the people have the same issue: <kbd>A</kbd> and <kbd>Shift+A</kbd> keys doesn't work. And those keys is a **part of `Atom` keyboard navigation** and part of `vscode-atom-keybindings` plugin as well.

**Comment by joaomoreno** [comment]
There was indeed an issue even when using the context key (which  https://github.com/Microsoft/vscode-atom-keybindings/pull/57 started using):

1. Open the Explorer, atom keybindings work
2. Switch to Search
3. Switch back to Explorer, atom keybindings don't work any more

Fixed by https://github.com/Microsoft/vscode/commit/5e54b2574b0703aa9437363294f8fd2db63f3e10

---

This is independent of your issue, @Ky6uk, since you do not seem to have any extension which sets that context key, so you need the setting.

**Comment by Ky6uk** [comment]
@Ky6uk That's okay if **my** issue will not be fixed yet, because I'm using `vscode-atom-keybindings`. It just was the same problem I reproduced with and without the extension. Will check it later on the next insider build.

**Comment by joaomoreno** [comment]
Tomorrow's insider will contain the setting for you: https://github.com/Microsoft/vscode/issues/68233

**Comment by Ky6uk** [comment]
@joaomoreno 

`VS Code version: Code - Insiders 1.32.0-insider (b64026839eed3dd03a6d6fe0ba3910255490c279`

This extension still doesn't work properly. 

Combinration <kbd>Cmd+Shift+E</kbd> (open explorer), then <kbd>A</kbd> (create new file) works only once. 
Second <kbd>A</kbd> ignored. 
Third <kbd>A</kbd> creates new file on the wrong level of the tree.

**Comment by joaomoreno** [comment]
@Ky6uk What do you do between <kbd>A</kbd> presses? Do you press <kbd>Escape</kbd>?

**Comment by Ky6uk** [comment]
@joaomoreno Yes. After <kbd>Esc</kbd> second <kbd>A</kbd> creates file on the wrong level (focus on the one directory, but file created on the other).

![bug](https://user-images.githubusercontent.com/46930/53245913-d9a69980-36b7-11e9-9fa1-6968cc59e9bb.gif)

**Comment by joaomoreno** [comment]
Right. Can you create a new issue in VS Code for that?
